src/artm/core/collection_parser.cc

```diff
@@ -139,7 +164,9 @@ void CollectionParser::ParseDocwordBagOfWordsUci(TokenMap* token_map) {
     if (item_id != prev_item_id) {
       prev_item_id = item_id;
       if (batch.item_size() >= config_.num_items_per_batch()) {
-        ::artm::core::BatchHelpers::SaveBatch(batch, config_.target_folder());
+        batch.set_id(boost::lexical_cast<std::string>(boost::uuids::random_generator()()));
+        ::artm::core::BatchHelpers::SaveBatch(batch, config_.target_folder(),
+          config_.name_type() == CollectionParserConfig_NameType_Guid ? batch.id() : batch_name_generator.next_name());
```

Comment:
Minor suggestion to remove `? :` from all `SaveBatch` calls:
* pass condition `config_.name_type() == CollectionParserConfig_NameType_Guid` to `BatchNameGenerator` constructor, and remember it inside `BatchNameGenerator` class
* add argument `const Batch& batch` to `BatchNameGenerator::next_name()`.